### PR TITLE
Unify widgets `shouldComponentUpdate` methods

### PR DESCRIPTION
**Pull request by victorporof**
Signed-off-by: Victor Porof <vporof@mozilla.com>

### CODE REVIEW

File: app/ui/shared/util/component-util.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// contain immutable.js props, which are throwing warnings when accessing some of
+// their properties while doing deep equality checks using lodash's `deepEqual`.
+//
+// See https://github.com/facebook/react/blob/master/src/addons/shallowCompare.js
+// for a similar implementation which assumes props are always immutable.js objects.
+//
+// This method is intended to be used only for components which don't contain
+// any immutable.js objects as props, only plain js objects. Simply use
+// React's `PureRenderMixin` directly when using only immutable.js props.
+// Don't mix immutable.js objects with plain js objects in props.
```
Comment (victorporof) [inline app/ui/shared/util/component-util.js]: Now with 100% more documentation.

File: app/ui/shared/util/component-util.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  const currImmutableProps = pick(this.props, immutables);
+  const nextImmutableProps = pick(nextProps, immutables);
+
+  const currPlainJsProps = omit(this.props, ['children', ...immutables]);
+  const nextPlainJsProps = omit(nextProps, ['children', ...immutables]);
+
+  return (
+    !shallowEqual(this.state, nextState)) ||
+    !shallowEqual(currChildren, nextChildren) ||
+    !shallowEqual(currImmutableProps, nextImmutableProps) ||
```
Comment (jsantell) [inline app/ui/shared/util/component-util.js]: Shouldn't we be using [Immutable.is](https://facebook.github.io/immutable-js/docs/#/is) for immutablesinstead of shallow equal?

Comment (victorporof) [inline app/ui/shared/util/component-util.js]: Absolutely not. Changed immutables are always new instances.

Comment (victorporof) [inline app/ui/shared/util/component-util.js]: See how PureRenderMixin works, I've linked to it in the comment

**Pull request merged**